src/string.cr

```diff
@@ -2005,13 +2010,34 @@ class String
   # "hello world".gsub('o', 'a') # => "hella warld"
   # ```
   def gsub(char : Char, replacement)
+    if replacement.is_a?(String) && replacement.bytesize == 1
+      return gsub(char, replacement.unsafe_byte_at(0).unsafe_chr)
+    end
+
     if includes?(char)
+      if replacement.is_a?(Char) && char.ascii? && replacement.ascii?
+        return gsub_ascii_char(char, replacement)
+      end
+
       gsub { |my_char| char == my_char ? replacement : my_char }
     else
       self
```

Comment:
Slightly unrelated but there is no direct coverage of this line in the specs. I wanted to double check the optimized overload have specs, and they do. But there is no `String#gsub(Char, replacement)` invocation that will stress this path. Other specs fails if this line is changed though.

Reply:
I'll add a spec.

Another discussion: `replacement` can be anything, like a number, array, string or char. Maybe we should limit it to only chars and strings? (like in Ruby)

Reply:
Done!

Reply:
I would leave it unbounded until there is another overload. I think is clear from the class and the name of the method what to expect.